### PR TITLE
fix: [2.4] Set corresponding DataScope for `loadStreamDelete`

### PR DESCRIPTION
**Pull request by congqixia**
Cherry-pick from master
pr: #35312
Related to #35311

### CODE REVIEW

File: internal/querynodev2/delegator/delegator_data.go [hunk 1]
```diff
@@ -658,7 +668,7 @@ func (sd *shardDelegator) loadStreamDelete(ctx context.Context,
 				SegmentId:    info.GetSegmentID(),
 				PrimaryKeys:  storage.ParsePrimaryKeys2IDs(deleteData.Pks),
 				Timestamps:   deleteData.Tss,
-				Scope:        querypb.DataScope_Historical, // only sealed segment need to loadStreamDelete
```
Comment (XuanYang-cn) [inline internal/querynodev2/delegator/delegator_data.go:661]: Why before L0 can we loadStreamDelete to Historical only? imo streamDelete should've been apply to all segments. will it affect 2.3?

Comment (congqixia) [reply internal/querynodev2/delegator/delegator_data.go:661]: before l0, binlog/deltalog and message from mq composes the whole growing segment

Comment (XuanYang-cn) [reply internal/querynodev2/delegator/delegator_data.go:661]: So loadStreamDelete process L0 segments only?

**Pull request merged**